### PR TITLE
fix: Corner cases for missing analyt. tables [DHIS2-15184]

### PR DESCRIPTION
**Pull request by maikelarabori**
This PR covers some corner cases scenarios regarding missing analytics tables.
Also adds e2e test coverage for the this scenario.
Some clean-up in the documentation and in the code are also present (concentrating the error handling in a single method).

### CODE REVIEW

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/JdbcEnrollmentAnalyticsManager.java [hunk 1]
```diff
@@ -254,27 +251,19 @@
...
-      }
-      if (!params.isMultipleQueries()) {
-        log.warn(AnalyticsUtils.ERR_MSG_SQL_SYNTAX_ERROR, ex);
-        throw ex;
-      }
-      log.warn(AnalyticsUtils.ERR_MSG_SILENT_FALLBACK, ex);
-    } catch (DataAccessResourceFailureException ex) {
-      log.warn(ErrorCode.E7131.getMessage(), ex);
-      throw new QueryRuntimeException(ErrorCode.E7131);
+    final String immutableValue = new String(sql);
```
[the remark this answers is not in the document]
Comment (maikelarabori) [reply dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/JdbcEnrollmentAnalyticsManager.java]: Done

**Review by gnespolino — APPROVED** [review]

**Review by d-bernat — APPROVED** [review]
